libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/couchdb/AttachmentStreamConnector.java

```diff
@@ -106,13 +107,25 @@ public InputStream unsafeGetAttachmentStream(AttachmentContent attachment) throw
         PipedOutputStream out = new PipedOutputStream(in);
 
         new Thread(() -> {
+            HashMap<String, Integer> addedFileNames = new HashMap<>();
             byte[] buffer = new byte[1024];
             int length;
 
             try(ZipOutputStream zip = new ZipOutputStream(out)){
                 for(AttachmentContent attachment : attachments) {
-                    // TODO: handle attachments with equal name
-                    ZipEntry zipEntry = new ZipEntry(attachment.getFilename());
+
+                    String originalFileName = attachment.getFilename();
+                    String acceptedFileName = originalFileName;
```

Comment:
Why not handle the result immediately in the corresponding if branch, instead of introducing this variable. This would reduce mutability.

Or maybe better: encapsulate this functionallity into its own function `String getAcceptedFileName(String originalFileName, Map<String, Integer> addedFileNames)`

Or maybe even better: `ZipEntry getDeduplicatedZipEntry(String originalFileName, Map<String, Integer> addedFileNames)`

Reply:
created a new method: `private ZipEntry getDuplicatedZipEntry(String fileName, Map<String, Integer> fileNameUsageMap)`

Reply:
I would expect `getDeduplicatedZipEntry`